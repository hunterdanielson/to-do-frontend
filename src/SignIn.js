import React, { Component } from 'react';
import request from 'superagent';
import './Common.css'

export default class SignIn extends Component {
    state = {
        email: '',
        password: ''
    }

    handleSubmit = async (e) => {
        e.preventDefault();
        try {

            const data = await request.post('https://sleepy-earth-23861.herokuapp.com/auth/signin', this.state)
            this.props.handleTokenChange(data.body.token, data.body.email);
            this.props.history.push('/quests');
        } catch {
            this.setState({ failure: 'login failed' })
        }
    }

    handleChange = (e) => {
        this.setState({ [e.target.name]: e.target.value })
    }

    render() {
        console.log(this.state)
        return (
            <div>
                <h1>Sign In</h1>
                <form onSubmit={this.handleSubmit}>
                    <label>
                        Email 
                        <input type='text' onChange={this.handleChange} name='email'></input>
                    </label>
                    <label>
                        Password
                        <input type='text' onChange={this.handleChange} name='password'></input>
                    </label>
                    <input type='submit'></input>
                </form>
                {this.state.failure && <p className='error'>Invalid email/password combo</p>}

                
            </div>
        )
    }
}
