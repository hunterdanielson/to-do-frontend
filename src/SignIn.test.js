import React from 'react';
import { shallow } from 'enzyme';
import SignIn from './SignIn.js';

test('renders signin correctly', () => {
  const wrapper = shallow(<SignIn />);
  expect(wrapper).toMatchSnapshot();
});