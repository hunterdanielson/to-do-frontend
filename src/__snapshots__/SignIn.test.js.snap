// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders signin correctly 1`] = `
<div>
  <h1>
    Sign In
  </h1>
  <form
    onSubmit={[Function]}
  >
    <label>
      Email
      <input
        name="email"
        onChange={[Function]}
        type="text"
      />
    </label>
    <label>
      Password
      <input
        name="password"
        onChange={[Function]}
        type="text"
      />
    </label>
    <input
      type="submit"
    />
  </form>
</div>
`;
